src/main/java/com/google/api/codegen/config/FlatteningConfig.java

```diff
@@ -41,12 +40,37 @@ public static FlatteningConfig createFlattening(
       DiagCollector diagCollector, FlatteningConfigProto flattening, Method method) {
     boolean missing = false;
     ImmutableList.Builder<ImmutableList<Field>> flatteningGroupsBuilder = ImmutableList.builder();
+    Set<Oneof> seenOneofs = new HashSet<>();
     for (FlatteningGroupProto flatteningGroup : flattening.getGroupsList()) {
       ImmutableList.Builder<Field> parametersBuilder = ImmutableList.builder();
       for (String parameter : flatteningGroup.getParametersList()) {
         Field parameterField = method.getInputMessage().lookupField(parameter);
         if (parameterField != null) {
           parametersBuilder.add(parameterField);
+          Oneof oneof = parameterField.getOneof();
+          if (oneof != null) {
+            diagCollector.addDiag(
+                Diag.warning(
+                    SimpleLocation.TOPLEVEL,
+                    "Oneof field flattened: method = %s, message type = %s, oneof = %s, field = %s",
+                    method.getFullName(),
+                    method.getInputMessage().getFullName(),
+                    oneof.getName(),
+                    parameter));
```

Comment:
Why be so permissive about the single flattened oneof parameter, if it's always an error for required parameters? This could be an opportunity to simplify the logic.